### ISSUE TITLE
InRange macro can trigger UB when it tries to negate MIN_INT

### ISSUE THREAD

**Issue by agroce** [trailofbits/deepstate#379]
Symptom (when compiling with ubsan):

```
TRACE: Running: binsearch_basic from test_binsearch.cpp(11)
TRACE: test_binsearch.cpp(16): a[0] = 757935405
/usr/local/include/deepstate/DeepState.h:391:1: runtime error: negation of -2147483648 cannot be represented in type 'int'; cast to an unsigned type to negate this value to itself
TRACE: test_binsearch.cpp(20): a[1] = 757935405
TRACE: test_binsearch.cpp(20): a[2] = 773140525
```

In rare but not impossible circumstances this can result in InRange returning a value that is NOT in range.  Not observed with clang++ compiling, but seen with afl-clang++ (interesting there is a difference here).  It's UB, and we need to account for that unusual possibility.

The relevant code is here:

https://github.com/trailofbits/deepstate/blob/0c66cfd934131419a9712cc955a4b12cd1dedd7d/src/include/deepstate/DeepState.h#L377

**Comment by agroce** [comment]
The code around there can also cause other UB:

```
WARNING: No seed provided; using 1618968284
WARNING: No test specified, defaulting to first test defined (InRange_InRange)
/usr/local/include/deepstate/DeepState.h:391:1: runtime error: signed integer overflow: 1987671424 - -1231949165 cannot be represented in type 'int'
```

**Comment by agroce** [comment]
Fixed in #380
